### PR TITLE
[#12048] Migrate enroll students action

### PR DESCRIPTION
**Pull request by domoberzin**
Part of #12048 

**Outline of Solution**
- Change `EnrollStudentsAction` to utilise SQL logic and database if course has been migrated
- Create methods to get/create Team and Section objects when necessary
- Modify and create associated methods for cascading update and creation in `UsersLogic`, `UsersDb`, `FeedbackResponsesLogic`, `FeedbackResponseCommentsLogic`, `DeadlineExtensionsLogic`
- Only added changes for non-email changes, since the enroll action did not allow for email changes in the original version
- Logic mirrors the original as closely as possible
- Created integration test for this action and modified the `TypicalDataBundle.json` file to suit test needs.

### CODE REVIEW

**Review by weiquu — CHANGES_REQUESTED** [review]
Hi @domoberzin, thanks for the PR, looks quite comprehensive so far!

Could you update the methods and action to use `Student` instead of `StudentAttributes`? The `*Attributes` classes are used for the datastore, so we shouldn't be using those in the migrated methods

File: src/main/java/teammates/sqllogic/api/Logic.java
Comment (weiquu) [inline src/main/java/teammates/sqllogic/api/Logic.java:211]: Should this be calling `.getSection` instead? Looks like a duplicate of `.getSectionOrCreate` below otherwise

File: src/main/java/teammates/sqllogic/core/UsersLogic.java
Comment (weiquu) [inline src/main/java/teammates/sqllogic/core/UsersLogic.java:835]: Is this method needed? Seems that it's only being called in `.getSectionForTeam` above, but that method isn't being called anywhere (it's only being used for datastore)

**Review by weiquu — COMMENTED** [review]
Left 2 preliminary comments! Will take a closer look later, but overall looks good

File: src/main/java/teammates/sqllogic/core/UsersLogic.java
Comment (domoberzin) [reply src/main/java/teammates/sqllogic/core/UsersLogic.java:835]: Thanks for catching, will remove both of these unnecessary methods

File: src/main/java/teammates/sqllogic/api/Logic.java
Comment (domoberzin) [reply src/main/java/teammates/sqllogic/api/Logic.java:211]: Will also add in a separate getSection method in the usersLogic level to resolve this

**Review by domlimm — COMMENTED** [review]
Really good work dom! Just a small thing you missed out.

File: src/main/java/teammates/sqllogic/core/UsersLogic.java
Comment (domoberzin) [reply src/main/java/teammates/sqllogic/core/UsersLogic.java]: Thanks for catching this, changes have been pushed

Comment (domlimm) [reply src/main/java/teammates/sqllogic/core/UsersLogic.java]: Good catch on the direct method for getting `getSection/TeamName`. I forgot about that 😓

**Review by domlimm — APPROVED** [review]
LGTM, nice work!

File: src/main/java/teammates/storage/sqlapi/DeadlineExtensionsDb.java
Comment (weiquu) [inline src/main/java/teammates/storage/sqlapi/DeadlineExtensionsDb.java:144]: Same comment as above

**Review by weiquu — COMMENTED** [review]
Should be final few comments! (We'll relook the `updateStudentCascade` method in another PR after confirming some edits to the database design)

File: src/main/java/teammates/common/datatransfer/attributes/StudentAttributes.java
Comment (domoberzin) [reply src/main/java/teammates/common/datatransfer/attributes/StudentAttributes.java]: Thanks for catching, will remove it as its unused anyway

**Comment by domlimm** [comment]
@domoberzin Looks really good! Thanks for being patient and making the changes promptly! Let's wait for wq to do the final review. Should be good to go! 🚀

**Review by weiquu — APPROVED** [review]
LGTM! Thanks for the quick fixes, great job

**Pull request merged**
